plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Main.java

```diff
@@ -1809,25 +1809,46 @@ private URL getConfigurationLocation() {
      */
     private URL buildProductURL() {
         String productConfigurationLocation = DBEAVER_CONFIGURATION_LOCTION;
-        String base = "";
-        URL url = null;
-        if (Constants.OS_WIN32.equals(getOS())) {
-            base = resolveEnv(productConfigurationLocation, DB_DATA_HOME, ENV_DATA_HOME_WIN);
-        } else if (Constants.OS_MACOSX.equals(getOS())) {
-            base = resolveLocation(productConfigurationLocation, DB_DATA_HOME, LOCATION_DATA_HOME_MAC);
-        } else {
-            base = resolveLocation(productConfigurationLocation, DB_DATA_HOME, LOCATION_DATA_HOME_UNIX);
-        }
+        String base = getWorkingDirectory(productConfigurationLocation);
         try {
             String productPath = getProductProperties();
             Path basePath = Paths.get(base, productPath);
             productConfigurationLocation = basePath.toFile().getAbsolutePath();
-            url = buildURL(productConfigurationLocation, true);
+            return buildURL(productConfigurationLocation, true);
         } catch (IOException e) {
             if (debug)
                 System.out.println("Can not read product properties. " + e.getMessage()); //$NON-NLS-1$
         }
-        return url;
+        return null;
+    }
+
+    public static String getWorkingDirectory(String defaultWorkspaceLocation) {
+        String osName = (System.getProperty("os.name")).toUpperCase();
+        String workingDirectory;
+        if (osName.contains("WIN")) {
+            String appData = System.getenv("AppData");
+            if (appData == null) {
+                appData = System.getProperty("user.home");
+            }
+            workingDirectory = appData + "\\" + defaultWorkspaceLocation;
+        } else if (osName.contains("MAC")) {
+            workingDirectory = System.getProperty("user.home") + "/Library/" + defaultWorkspaceLocation;
+        } else {
+            // Linux
+            String dataHome = System.getProperty("XDG_DATA_HOME");
+            if (dataHome == null) {
+                dataHome = System.getProperty("user.home") + "/.local/share";
+            }
+            String badWorkingDir = dataHome + "/." + defaultWorkspaceLocation;
+            String goodWorkingDir = dataHome + "/" + defaultWorkspaceLocation;
+            if (!new File(goodWorkingDir).exists() && new File(badWorkingDir).exists()) {
+                // Let's use bad working dir if it exists (#6316)
+                workingDirectory = badWorkingDir;
+            } else {
+                workingDirectory = goodWorkingDir;
+            }
+        }
+        return workingDirectory;
```

Comment:
It looks like a duplicate of RuntimeUtils::getWorkingDirectory

Reply:
It is duplicate. It is the main idea - to use exactly the same algorithm as in DBeaver itself.
We cannot reuse this function because launcher doesn't have access to dbeaver model (to any other jar actually)